tools/codegen/api/python.py

```diff
@@ -710,11 +698,9 @@ def signature(f: NativeFunction, *, method: bool = False, pyi: bool = False) ->
             default='None' if pyi else _dtype_default_type_hack(name),
             default_init='self.scalar_type()' if is_like_or_new_function else None,
         ))
-        # TODO: probably a bug, kill this diff?
-        # pyi signatures have a slightly different type/default for layout
         tensor_options_args.append(PythonArgument(
             name='layout',
-            type=BaseType(BaseTy.Layout) if pyi else OptionalType(BaseType(BaseTy.Layout)),
+            type=OptionalType(BaseType(BaseTy.Layout)),
             default='strided' if pyi else 'torch.strided',
```

Comment:
On second look, I think python binding code is actually wrong here and pyi is right (it should be `strided`).

All other uses of classes in the torch namespace are explicitly imported into the pyi file. For example, we use `Tensor` and not `torch.Tensor` throughout both the pyi and python binding code.

The python_arg_parser is just hardcoded to pick up `torch.strided`, so it would be a pretty easy change to unify them to use `strided`. Probably best for a separate PR that cleans up python binding hacks though.